### PR TITLE
improvement(events tab): show nemesis that happened

### PR DESCRIPTION
**Pull request by temichus**
show which nemesis ran during each event
on events tab

Screenshot from 2023-04-27 11-21-18
Screenshot from 2023-04-27 11-21-09

### CODE REVIEW

File: frontend/TestRun/EventsTab.svelte [hunk 1]
```diff
@@ -117,11 +117,29 @@
             return parsed;
         });
 
+        let sortedNemesises = nemesisData.sort((a, b) => a.start_time - b.start_time);
+        let nemesisIndex = 0;
+        for (let event of parsedEvents.sort((a, b) => a.time - b.time)) {
+            event.nemesis = undefined;
+            if ( sortedNemesises[nemesisIndex]) {
+                while(sortedNemesises[nemesisIndex]){
```
Comment (aleksbykov) [inline frontend/TestRun/EventsTab.svelte:125]: Do you need outer if statement? if sortedNemesises[nemesisInded] is false whille loop also will not run

**Review by aleksbykov — APPROVED** [review]
LGTM, Small question

File: frontend/TestRun/EventsTab.svelte [hunk 2]
```diff
@@ -117,11 +117,29 @@
             return parsed;
         });
 
+        let sortedNemesises = nemesisData.sort((a, b) => a.start_time - b.start_time);
+        let nemesisIndex = 0;
+        for (let event of parsedEvents.sort((a, b) => a.time - b.time)) {
+            event.nemesis = undefined;
+                while(sortedNemesises[nemesisIndex]){
```
Comment (k0machi) [inline frontend/TestRun/EventsTab.svelte:124]: You should correct the indentation of the while block and also fix spaces before `{` and after `}` characters (there should be a space in every case, e.g. `while (true) {`)

File: frontend/TestRun/EventsTab.svelte [hunk 3]
```diff
@@ -117,11 +117,29 @@
...
+                    }else{
+                        nemesisIndex += 1;
+                    }
+                }
+
+        }
+
         return parsedEvents.sort((a, b) => b.time - a.time);
     };
+    0
```
Comment (k0machi) [inline frontend/TestRun/EventsTab.svelte:139]: Looks like a typo here

File: frontend/TestRun/EventsTab.svelte [hunk 4]
```diff
@@ -117,11 +117,29 @@
...
+        let nemesisIndex = 0;
+        for (let event of parsedEvents.sort((a, b) => a.time - b.time)) {
+            event.nemesis = undefined;
+                while(sortedNemesises[nemesisIndex]){
+                    if (event.time.getTime() < sortedNemesises[nemesisIndex].start_time * 1000) {
+                        break;
+                    }else if(event.time.getTime() <= sortedNemesises[nemesisIndex].end_time * 1000){
+                        event.nemesis = sortedNemesises[nemesisIndex];
+                        break;
+                    }else{
```
Comment (k0machi) [inline frontend/TestRun/EventsTab.svelte:130]: The `else` not needed as we break out in every other conditional, can just `nemesisIndex++` here.

File: frontend/TestRun/StructuredEvent.svelte [hunk 5]
```diff
@@ -27,8 +27,13 @@
 >
     <div class="event-header d-flex align-items-start flex-wrap">
         <div class="ms-2 mb-2 bg-dark text-light rounded px-2">{event.eventType}</div>
-        <div class="ms-2 mb-2 me-auto rounded px-2 severity-{event.severity.toLowerCase()}">{event.severity}</div>
-        <div class="ms-2 mb-2 rounded px-2">
+        <div class="ms-2 mb-2 rounded px-2 severity-{event.severity.toLowerCase()}">{event.severity}</div>
+        {#if event.nemesis}
+        <div class="ms-2 mb-2 rounded px-2 status-{event.nemesis.status.toLowerCase()}">{event.nemesis.name}</div>
```
Comment (k0machi) [inline frontend/TestRun/StructuredEvent.svelte:32]: The `div`s inside conditional blocks should be embedded.

**Review by k0machi — CHANGES_REQUESTED** [review]
Indendation for the while block is now offset, and some small comments about style.

**Review by roydahan — APPROVED** [review]

**Pull request merged**